fiftyone/core/stages.py

```diff
@@ -1002,7 +1002,7 @@ class SelectFields(ViewStage):
     """
 
     def __init__(self, field_names=None):
-        default_fields = default_sample_fields(include_private=False)
+        default_fields = default_sample_fields(include_private=True)
```

Comment:
This broke a unit test:
```
______________________ ViewStageTests.test_select_fields _______________________

self = <unittests.ViewStageTests testMethod=test_select_fields>

    def test_select_fields(self):
        self.dataset.add_sample_field("select_fields_field", fo.IntField)
    
        for sample in self.dataset.select_fields():
>           self.assertSetEqual(
                sample.selected_field_names, set(default_sample_fields())
            )
E           AssertionError: Items in the first set but not the second:
E           '_rand'
```
Unsure if this or the test needs to be changed.

Reply:
We need `_id` returned, but not `_rand`, etc. `include_private` is too general

Reply:
I'll push a fix, one moment

Reply:
No `_rand` needs to be included here. If you omit `_rand` when a `SelectFields` stage is included, then it will break pipelines that intend to include a `Take` or `Shuffle` stage after that.

I will handle fixing this. Working on other things at the moment, which is why I haven't acted on his PR yet

Reply:
:+1: